test/unit/services/matchSurnameToAppealTest.js

```diff
@@ -0,0 +1,112 @@
+const { matchSurnameToAppeal } = require('app/services/matchSurnameToAppeal');
+const {expect, sinon} = require('test/chai-sinon');
+const nock = require('nock');
+const { appealsAPI } = require('app/config');
+const { fail } = require('assert');
+const HttpStatus = require('http-status-codes');
+const validateSurname = require('app/assets/locale/en').validateSurname;
+
+describe('matchSurnameToAppeal.js', () => {
+
+  const invalidmactoken = 'invalidToken';
+  const invalidSurname = 'invalidSurname';
+  const appealId = '1234';
+  const errorFields = {
+    error: true,
+    surname: {
+      value: invalidSurname,
+      error: true,
+      errorMessage: validateSurname.surname.errors.noMatch,
+      errorHeading: validateSurname.surname.errors.noMatch
+    }
+  };
+  let req, res, next, api;
+
+  beforeEach(() => {
+    req = {
+      params: {},
+      body: {}
+    };
+    res = {
+      redirect: sinon.stub(),
+      locals: {
+        appeal: {},
+        i18n: {
+          validateSurname
+        }
+      }
+    };
+    next = sinon.stub();
+  });
+
+  afterEach(() => {
+    api.isDone();
+  });
+
+  describe('get request to api is successful', ()=> {
+
+    it('should call res.redirect', () => {
+      req.params.mactoken = 'validToken';
+      req.body.surname = 'validSurname';
+      api = nock(appealsAPI)
+        .get(`/validate/${req.params.mactoken}/${req.body.surname}`)
+        .reply(200, { appealId });
+
+      return matchSurnameToAppeal(req, res, next)
+        .then(() => {
+          res.redirect.should.have.been.calledWith(`/progress/${appealId}/trackyourappeal`);
+        })
+        .catch(err => {
```

Comment:
I don't understand why we have defined a catch block when the API is mocked to respond with a HTTP 200? Removing the catch block the test still passes.